File: middleware-demo/typescript/functions/hello.ts
import { api } from "@nitric/sdk";

const validate = (ctx, next) => {
  // Perform request validation, etc.
  console.log("This is  your middleware function running ...");
  next(ctx);
};

const helloApi = api("hello", {
  middleware: [validate],
});

helloApi.get("/hello/:name", async (ctx) => {
  const { name } = ctx.req.params;
  console.log("This is  your main function running ...");
  ctx.res.body = `Hello ${name}`;

  return ctx;
});

File: middleware-demo/python/services/hello.py
from nitric.application import Nitric
from nitric.resources import api, ApiOptions

from middle import run_first

publicApi = api("public", ApiOptions(
    middleware=[run_first],
))


@publicApi.get("/hello")
async def get_profiles(ctx):
    print("This is  your main function running ...")

    ctx.res.body = "hello"


Nitric.run()
